### PR TITLE
Separate services

### LINKED ISSUE #288: Split storage and encryption service
# Description
The storage and encryption service should serve as separate implementations on top of the encryptonize library. We should therefore split them into independent services. Care should be taken to do this in a way where we can reuse common logic (user management, etc.).

### PR DESCRIPTION
**Pull request by philip-cc**
This PR does a major restructuring of core to anticipate our new strategy of building various "services" on top of libencryptonize. The idea is to have an `integrations` folder (better name suggestions welcome) where the various services live, as well as shared service code. Currently we have two services: Blob and EaaS. 

The new folder structure looks like this:
```
├── client                 # Client code. Currently shared between blob and eaas
├── documentation          # User docs
├── e2etests               # E2E tests for the integrations
├── integrations           # This is where all the services live. Better name welcome
│   ├── blob               # Encryptonize Blob, formerly known as the storage service
│   ├── common             # All code shared by the services: context, shared handlers, middleware, storage, etc.
│   ├── eaas               # Encryptonize EaaS, formerly known as the encryption service
│   └── protobuf           # All protobuf files live here
├── kubernetes             # Kubernetes files. These are not up to date.
└── lib                    # The encryptonize library
```

I have tried to make the coupling a bit looser than it previously was: 
* The code in `integrations/common` should be independent of the services, i.e. the services consume stuff in `common`.
* The blob and eaas services are independent go modules.
* The protobufs are their own go module.
* E2E tests are their own module that simply uses the go client. 

### Relevant Issues/PRs
Closes #288 

### Type(s) of Change (Split the PR if you check many)
- [ ] Added user feature
- [ ] Added technical feature
- [ ] Added tests
- [x] Refactor
- [ ] Bugfix
- [ ] Updated documentation
- [ ] Kubernetes changes
- [ ] CI/CD workflow changes

### Testing Checklist
- [x] I have added/updated unit tests for functional changes.
- [x] I have added/updated integration tests for changes that affect dependent systems.
- [x] I have added/updated end-to-end tests for feature changes.

### General Checklist
- [x] I have pulled in the latest changes from master.
- [x] I have run `make lint`.
- [x] I have successfully run `make tests`.
- [ ] I have updated relevant CI/CD workflows.
- [ ] I have updated relevant Kubernetes manifests/scripts.
- [ ] I have updated/added relevant (internal and external) documentation (readme, doc comments, etc).
- [ ] I have updated the dependency map to reflect my changes.
- [ ] I have added the license to new source code files.
- [x] I have spent some time looking over the full diff before creating this PR.

### Technical Debt
#303 
https://github.com/cyber-crypt-com/documentation/issues/127
https://github.com/cyber-crypt-com/encryptonize-premium/issues/157
#304
#305

### CODE REVIEW

File: integrations/common/handlers/user_handler.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+	if err != nil {
+		log.Fatal(ctx, err, "Could not generate uuid")
+	}
+	ctx = icontext.CtxWithRequestID(ctx, requestID)
+
+	authStoreTxCreate, err := authStore.NewTransaction(ctx, requestID)
+	if err != nil {
+		log.Fatal(ctx, err, "Authstorage Begin failed")
+	}
+	defer func() {
```
Comment (kim-cc) [inline integrations/common/handlers/user_handler.go:102]: Maybe add a comment here describing the reason for the rollback. It looks weird that a rollback is always performed, but it makes sense when you look at `NewCreateUserHandler`

File: .github/workflows/snyk-scan.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v2
+
+    - name: Setup and cache dependencies
+      uses: cyber-crypt-com/encryptonize-core/.github/actions/encryption-service-build-setup@master
+
+    - name: Build Encryptonize Blob Docker image
+      working-directory: ./integrations/blob
```
Comment (kim-cc) [inline .github/workflows/snyk-scan.yaml:37]: shouldn't this be `objects` instead of `blob`?

Comment (philip-cc) [reply .github/workflows/snyk-scan.yaml:37]: Yes, workflows are work-in-progress.

File: integrations/common/handlers/user_handler.go (hunk 1 above)
Comment (philip-cc) [reply integrations/common/handlers/user_handler.go]: I have rewritten this to use the middlewares instead. Should make it more obvious what's going on and less error prone.

File: .github/workflows/snyk-scan.yaml (hunk 2 above)
Comment (philip-cc) [reply .github/workflows/snyk-scan.yaml]: Workflows should be up to date now.

**Review by kim-cc — DISMISSED** [review]

**Review by claudiu-cc — APPROVED** [review]

**Pull request merged**